src/services/review/RepoCheckerService.ts

```diff
@@ -0,0 +1,234 @@
+        files.forEach((file) => {
+          mdFiles.add(file)
+        })
+        console.log("calling getAllDocumentsPath", repoPath)
+        return this.getAllMediaPath(repoPath)
+      })
+      .andThen((setOfAllMediaPath) => {
+        console.log(setOfAllMediaPath)
+
+        return ResultAsync.fromPromise(
+          this.getAllPermalinks(repoPath, mdFiles),
+          (error) => new SiteCheckerError(`${error}`)
+        ).andThen((setOfAllPermalinks) =>
+          ok(new Set([...setOfAllMediaPath, ...setOfAllPermalinks]))
+        )
+      })
+      .andThen((setOfAllMediaAndPagesPath) => {
+        mdFiles.forEach(async (fileName) => {
+          const file = fs.readFileSync(path.join(repoPath, fileName), "utf8")
+          const filePermalink = file
+            .split("---")[1]
+            .split("permalink: ")[1]
+            .split("\n")[0]
+          const fileContent = file.split("---")[2]
+          if (!fileContent) {
+            return
+          }
+          const html = await marked.parse(fileContent)
+          const dom = new JSDOM(html)
+          const anchorTags = dom.window.document.querySelectorAll("a")
+          forEach(anchorTags, (tag) => {
+            const href = tag.getAttribute("href")
+            const text = tag.textContent
+            if (href) {
+              console.log(href)
+              const encodedHref = this.documentPathDecoder(href)
+              if (
+                !this.isExternalLinkOrPageRef(encodedHref) &&
+                !setOfAllMediaAndPagesPath.has(encodedHref) &&
+                // Amplify supports adding of trailing slash
+                !setOfAllMediaAndPagesPath.has(`${encodedHref}/`)
+              ) {
+                console.log(setOfAllMediaAndPagesPath)
+                this.persistLogs(repo, [
+                  new SiteCheckerError(
+                    `Broken link ${href} that links ${text} found in ${fileName} which has permalink ${filePermalink}`
+                  ),
+                ])
+              }
+            }
+          })
+          const imgTags = dom.window.document.querySelectorAll("img")
+          forEach(imgTags, (tag) => {
+            const src = tag.getAttribute("src")
+            if (src) {
+              const encodedSrc = this.documentPathDecoder(src)
+              if (
+                !this.isExternalLinkOrPageRef(encodedSrc) &&
+                !setOfAllMediaAndPagesPath.has(encodedSrc)
+              ) {
+                this.persistLogs(repo, [
+                  new SiteCheckerError(
+                    `Broken image ${src} found in ${fileName} which has permalink ${filePermalink}`
+                  ),
+                ])
+              }
+            }
+          })
+        })
+        console.log("Site checker done")
+        return ok(undefined)
+      })
+    try {
+      console.log(await checker)
+    } catch (error) {
+      console.log(error)
+    }
+  }
+
+  isExternalLinkOrPageRef(link: string) {
```

Comment:
Assuming this is fixed upstream? Also, the `www` check is wrong and might need removal

Reply:
arh will introduce schema upstream. for now just use starts with "/" then.

Reply:
Hmm, the problem is not with this specific error, but more that our link checker misses out some errors, which are caught by off the shelf tools

I think, regardless, this pr helps to feedback broken links so all's good but it doesn't bode well that we have missing broken links right off the bat.

Tbh, no issue with this going into production but we do have to be clear that:
1. Our efforts are better off identifying links that are not broken, but redirect to an internal facing site (eg: prod links going to staging) 
2. We'd have to relook at this post rollout to decide what forms of broken links we want to catch.